packages/es-components/src/components/controls/buttons/SpinnerButton.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from './Button';
+import Spinner from '../../base/spinner/Spinner';
+
+const DISPLAY_TYPES = {
+  none: 'none',
+  inline: 'inline'
+};
+
+function SpinnerButton({
+  children,
+  onClick,
+  styleType,
+  size,
+  block,
+  mobileBlock,
+  flatLeftEdge,
+  flatRightEdge,
+  title,
+  description,
+  ...other
+}) {
+  const [displaySpinner, setDisplay] = useState(DISPLAY_TYPES.none);
+
+  const defaultDimensions = {
+    maxWidth: 100,
+    height: 40
+  };
+
+  const loadSpinnerTriggerOnClick = () => {
+    onClick();
+    setDisplay(DISPLAY_TYPES.inline);
+
+    setTimeout(() => {
+      setDisplay(DISPLAY_TYPES.none);
+    }, 2000);
```

Comment:
wouldn't you want to pass an outside prop that tells when the button spins or not

Reply:
I did consider that. When I looked at how we're using buttons in our MFE and others I saw that we're pretty much using them for progression throughout apps and once we've left a page the button will stop spinning and doesn't leak or anything.

Reply:
@Zlasa I wrapped the onClick in a promise and shortened the timeout to 500ms. If there isn't some sort of delay from the onClick event the spinner won't display.

Reply:
what happens if the request takes a sec for example? that's why i think passing in a variable is a way to go

Reply:
I too wonder if it might be simpler and more flexible if the component just took in a prop like `showSpinner`. It should be pretty easy to control externally, much like a prop to show a modal.

If we can do that, I actually don't see why we couldn't just add the spinner to the regular Button component since all it would be is the additional `div` that's conditionally displayed.

Reply:
So I just read Steve's comments and Stewart's. This looks to be the consensus. Stewart's draft PR's are accomplishing this already without adding a new component.